### ISSUE TITLE
Using only symbols are not RTL when running the project with internationalization/locale/test=ar

### ISSUE THREAD

**Issue by dhoverml** [godotengine/godot#81348]
### Godot version

v4.2.dev4.official [549fcce5f]

### System information

Ubuntu 20.04 LTS

### Issue description

See title.

### Steps to reproduce

1. Create a new project
2. Create a csv file that includes the following
```
key,ar
key,placeholder
```
3. Add the generated `.translation` file to Project -> Project Settings... -> Localization
4. In Project -> Project Settings... -> Internationalization -> Locale, set `Test` to `ar`
5. Add a `LineEdit` node and type `-=[]{}`
6. Run the project, and notice that the text is on the right side, but the symbols are incorrectly in LTR (`-=[]{}`).  Correct output is `{}[]=-`
7. Back in the editor, add a single Arabic letter (any will do), and notice that `LineEdit` immediately (and correctly) flips the symbols, and the newly-added Arabic letter, to RTL.

**Notes**
- Identical behavior is seen with `TextEdit`.
- The purpose of steps 2-4 are to simulate an Arabic environment when the project is run, despite `OS.get_locale() == "en"` (or your system's current language).

### Minimal reproduction project

[rtlsymbols.zip](https://github.com/godotengine/godot/files/12525509/rtlsymbols.zip)

**Comment by akien-mga** [comment]
CC @bruvzg

**Comment by bruvzg** [comment]
It's not a bug, by default text direction is determined only based on the text itself, if you want it to be based on locale/layout direction you can switch text direction from "Auto" to "Inherited". But detection for neutral text probably can be improved.

<img width="371" alt="Screenshot 2023-09-06 at 08 33 58" src="https://github.com/godotengine/godot/assets/7645683/9ba55929-bdd4-4b48-aea6-f07a878dc562">

**Comment by dhoverml** [comment]
Thanks @bruvzg !  I tested your fix and it resolves this bug.

However I see two possible, unrelated, minor bugs with both your fix and 4.1.1-stable:

First:
- (good) setting `Text Direction` to `Inherited` for `LineEdit` works.
- (bad) setting `Text Direction` to `Inherited` for `TextEdit` also works, however I see `Invalid text direction` when I set the setting in the editor, and a bunch of these when I run the project:
![text_edit_text_direction_inherited](https://github.com/godotengine/godot/assets/135338122/544a2711-7dd9-4199-8641-66d96e09907e)

Second:
The text and cursor slightly shifts left and right as I type any character.  I am able to repro with:
1. Create a new project
2. Add a LineEdit node
3. Set `Layout Direction` to `Right-to-Left`
4. Run the project
5. (bad) Type `ababab`, one at a time, and notice the cursor shifts slightly to the left and right as you type `a` then `b`.  It's subtle.
6. (good) Back in the editor, set `Font size` to `50`.  Relaunch the app, and notice typing `ababab` doesn't shift the cursor as you type each letter.

Let me know if you'd like separate issues written for one or both of these.
